lumeer-core/src/main/java/io/lumeer/core/facade/AbstractFacade.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *  
+ * Copyright (C) since 2016 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.core.facade;
+
+import io.lumeer.api.model.Permission;
+import io.lumeer.api.model.Permissions;
+import io.lumeer.api.model.Resource;
+import io.lumeer.api.model.Role;
+import io.lumeer.core.AuthenticatedUser;
+import io.lumeer.core.PermissionsChecker;
+import io.lumeer.core.model.SimplePermission;
+
+import java.util.Set;
+import javax.inject.Inject;
+
+public abstract class AbstractFacade {
+
+   @Inject
+   protected PermissionsChecker permissionsChecker;
+
+   @Inject
+   protected AuthenticatedUser authenticatedUser;
+
+   public Resource keepOnlyActualUserRoles(final Resource resource) {
```

Comment:
You can make use of generic types here in order to avoid casting when using this method.
```java
public <T extends Resource> T keepOnlyActualUserRoles(final T resource) {
   ...
}
```